### PR TITLE
Refactor code with PEP8 and additional limitations. Fix #1521

### LINKED ISSUE #1521: Refactor current gensim code by PEP8
We have a lot of old code which doesn't satisfy PEP8, it's time to fix it!

What we want:
- [x] PEP8 (without a hard 80-char constraint for long lines)
- [x] No vertical indents (only hanging)
- [x] map/filter -> list/set/dict comprehension
- [x] logging message arguments as method arguments (not direct formatting)

### PR DESCRIPTION
**Pull request by menshikh-iv**

### CODE REVIEW

File: gensim/models/wrappers/ldamallet.py [hunk 1]
```diff
@@ -305,14 +305,9 @@ def read_doctopics(self, fname, eps=1e-6, renorm=True):
...
-                    doc = [(id_, weight)
-                           for id_, weight in zip(map(int, parts[::2]),
-                                                  map(float, parts[1::2]))
-                           if abs(weight) > eps]
+                    doc = [(id_, weight) for id_, weight in ((int(x), float(y)) for (x, y) in zip(parts[::2], parts[1::2])) if abs(weight) > eps]
                 elif len(parts) == self.num_topics and mallet_version != '2.0.7':
-                    doc = [(id_, weight)
-                           for id_, weight in enumerate(map(float, parts))
-                           if abs(weight) > eps]
+                    doc = [(id_, weight) for id_, weight in enumerate(float(x) for x in parts) if abs(weight) > eps]
```
Comment (zsef123) [inline gensim/models/wrappers/ldamallet.py]: @menshikh-iv 
line 308:
```
doc = [(int(id_), float(weight))
           for id_, weight in zip(*[iter(parts)] * 2)
           if abs(float(weight)) > eps]
```

line 310 :
```
doc = [(id_, float(weight))
           for id_, weight in enumerate(parts)
           if abs(float(weight)) > eps]
```
How about above code?
Above code is less loop

File: gensim/test/test_sklearn_api.py [hunk 2]
```diff
@@ -363,7 +363,7 @@ def testPipeline(self):
         data = cache
         test_data = data.data[0:2]
         test_target = data.target[0:2]
-        id2word = Dictionary(map(lambda x: x.split(), test_data))
+        id2word = Dictionary([x.split() for x in data.data])
```
Comment (zsef123) [inline gensim/test/test_sklearn_api.py]: `data.data` need change to `test_data`

File: gensim/test/test_atmodel.py [hunk 3]
```diff
@@ -436,7 +436,7 @@ def testPasses(self):
             for test_rhot in test_rhots:
                 model.update(corpus, author2doc)
 
-                msg = ", ".join(map(str, [passes, model.num_updates, model.state.numdocs]))
+                msg = ", ".join(str(x) for x in [passes, model.num_updates, model.state.numdocs])
```
Comment (zsef123) [inline gensim/test/test_atmodel.py]: How about `msg = "%d, %d, %d" % (passes, model.num_updates, model.state.numdocs)`?
This is more simple.

File: gensim/test/test_sklearn_api.py (hunk 2 above)
Comment (menshikh-iv) [reply gensim/test/test_sklearn_api.py]: Thanks, fixed.

File: gensim/test/test_atmodel.py (hunk 3 above)
Comment (menshikh-iv) [reply gensim/test/test_atmodel.py]: Agree, fixed

File: gensim/models/wrappers/ldamallet.py (hunk 1 above)
Comment (menshikh-iv) [reply gensim/models/wrappers/ldamallet.py]: Ok, accepted.

Comment (piskvorky) [reply gensim/models/wrappers/ldamallet.py]: @zsef123 Both examples above are incorrect. The correct formatting using hanging indent:

```python
doc = [
    (id_, float(weight))
    for id_, weight in enumerate(parts)
    if abs(float(weight)) > eps]
```

or (I personally prefer this):

 ```python
doc = [
    (id_, float(weight))
    for id_, weight in enumerate(parts)
    if abs(float(weight)) > eps
]
```

Actually, this line not too long, so simple:

```python
doc = [(id_, float(weight)) for id_, weight in enumerate(parts) if abs(float(weight)) > eps]
```

works too.

File: gensim/examples/dmlcz/dmlcorpus.py [hunk 4]
```diff
@@ -105,22 +105,20 @@ def buildDictionary(self):
...
         """
-        logger.info("creating dictionary from %i articles" % len(self.documents))
+        logger.info("creating dictionary from %i articles", len(self.documents))
         self.dictionary = dictionary.Dictionary()
         numPositions = 0
         for docNo, (sourceId, docUri) in enumerate(self.documents):
             if docNo % 1000 == 0:
-                logger.info("PROGRESS: at document #%i/%i (%s, %s)" %
-                             (docNo, len(self.documents), sourceId, docUri))
+                logger.info("PROGRESS: at document #%i/%i (%s, %s)", docNo, len(self.documents), sourceId, docUri)
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/examples/dmlcz/dmlcorpus.py]: Completely agree, will removed it soon (but not in this PR)

File: gensim/corpora/sharded_corpus.py [hunk 5]
```diff
@@ -254,20 +252,14 @@ def init_shards(self, output_prefix, corpus, shardsize=4096, dtype=_default_dtyp
         """Initialize shards from the corpus."""
 
         if not gensim.utils.is_corpus(corpus):
-            raise ValueError('Cannot initialize shards without a corpus to read'
-                             ' from! (Got corpus type: {0})'.format(type(corpus)))
+            raise ValueError('Cannot initialize shards without a corpus to read from! (Got corpus type: {0})'.format(type(corpus)))
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/sharded_corpus.py]: For my opinion both variants is good, for this reason we shouldn't use only one

File: gensim/models/wrappers/ldavowpalwabbit.py [hunk 6]
```diff
@@ -140,11 +140,8 @@ def __init__(self, vw_path, corpus=None, num_topics=100, id2word=None,
 
         if self.id2word is None:
             if corpus is None:
-                raise ValueError('at least one of corpus/id2word must be '
-                                 'specified, to establish input space '
-                                 'dimensionality')
-            LOG.warning('no word id mapping provided; initializing from '
-                        'corpus, assuming identity')
+                raise ValueError('at least one of corpus/id2word must be specified, to establish input space dimensionality')
+            LOG.warning('no word id mapping provided; initializing from corpus, assuming identity')
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/models/wrappers/ldavowpalwabbit.py]: Fixed.

File: gensim/corpora/dictionary.py [hunk 7]
```diff
@@ -333,7 +322,7 @@ def merge_with(self, other):
             old2new[other_id] = new_id
             try:
                 self.dfs[new_id] += other.dfs[other_id]
-            except Exception:
+            except AttributeError:
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/dictionary.py]: Fixed

File: gensim/models/wrappers/ldamallet.py (hunk 1 above)
Comment (menshikh-iv) [reply gensim/models/wrappers/ldamallet.py]: already done as your last variant @piskvorky

File: gensim/corpora/dictionary.py [hunk 8]
```diff
@@ -244,20 +241,12 @@ def filter_tokens(self, bad_ids=None, good_ids=None):
         """
         if bad_ids is not None:
             bad_ids = set(bad_ids)
-            self.token2id = dict((token, tokenid)
-                                 for token, tokenid in iteritems(self.token2id)
-                                 if tokenid not in bad_ids)
-            self.dfs = dict((tokenid, freq)
-                            for tokenid, freq in iteritems(self.dfs)
-                            if tokenid not in bad_ids)
+            self.token2id = dict((token, tokenid) for token, tokenid in iteritems(self.token2id) if tokenid not in bad_ids)
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/dictionary.py]: Fixed

File: gensim/corpora/sharded_corpus.py [hunk 9]
```diff
@@ -484,8 +456,7 @@ def resize_shards(self, shardsize):
             # If something happens when we're in this stage, we're screwed.
             except Exception as e:
                 print(e)
-                raise RuntimeError('Resizing completely failed for some reason.'
-                                   ' Sorry, dataset is probably ruined...')
+                raise RuntimeError('Resizing completely failed for some reason. Sorry, dataset is probably ruined...')
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/sharded_corpus.py]: Fixed

File: gensim/corpora/sharded_corpus.py [hunk 10]
```diff
@@ -529,22 +500,13 @@ def _guess_n_features(self, corpus):
                 return self._guess_n_features(corpus.corpus)
         else:
             if not self.dim:
-                raise TypeError('Couldn\'t find number of features, '
-                                 'refusing to guess (dimension set to {0},'
-                                 'type of corpus: {1}).'.format(self.dim, type(corpus)))
+                raise TypeError('Couldn\'t find number of features, refusing to guess (dimension set to {0}, type of corpus: {1}).'.format(self.dim, type(corpus)))
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/sharded_corpus.py]: Fixed  (`\'` occurs only in this file several times)

File: gensim/corpora/dictionary.py [hunk 11]
```diff
@@ -201,15 +199,14 @@ def filter_extremes(self, no_below=5, no_above=0.5, keep_n=100000, keep_tokens=N
...
         good_ids = sorted(good_ids, key=self.dfs.get, reverse=True)
         if keep_n is not None:
             good_ids = good_ids[:keep_n]
-        bad_words = [(self[id], self.dfs.get(id, 0)) for id in set(self).difference(good_ids)]
+        bad_words = [(self[idx], self.dfs.get(idx, 0)) for idx in set(self).difference(good_ids)]
         logger.info("discarding %i tokens: %s...", len(self) - len(good_ids), bad_words[:10])
-        logger.info(
-            "keeping %i tokens which were in no less than %i and no more than %i (=%.1f%%) documents",
-            len(good_ids), no_below, no_above_abs, 100.0 * no_above)
+        logger.info("keeping %i tokens which were in no less than %i and no more than %i (=%.1f%%) documents", len(good_ids), no_below, no_above_abs, 100.0 * no_above)
```
[the remark this answers is not in the document]
Comment (menshikh-iv) [reply gensim/corpora/dictionary.py]: Fixed

**Pull request merged**